includes/log-handlers/class-wc-log-handler-email.php

```diff
@@ -0,0 +1,185 @@
+
+	/**
+	 * Minimum log level this handler will process.
+	 *
+	 * @var int Integer representation of minimum log level to handle.
+	 */
+	protected $threshold;
+
+	/**
+	 * Stores email recipients.
+	 *
+	 * @var array
+	 */
+	protected $recipients = array();
+
+	/**
+	 * Stores log messages.
+	 *
+	 * @var array
+	 */
+	protected $logs = array();
+
+	/**
+	 * Constructor for log handler.
+	 *
+	 * @param string|array $recipients Optional. Email(s) to receive log messages. Defaults to site admin email.
+	 * @param string $threshold Optional. Minimum level that should receive log messages.
+	 *     Default 'alert'. One of: emergency|alert|critical|error|warning|notice|info|debug
+	 */
+	public function __construct( $recipients = null, $threshold = 'alert' ) {
+		if ( null === $recipients ) {
+			$recipients = get_option( 'admin_email' );
+		}
+
+		if ( is_array( $recipients ) ) {
+			foreach ( $recipients as $recipient ) {
+				$this->add_email( $recipient );
+			}
+		} else {
+			$this->add_email( $recipients );
+		}
+
+		$this->set_threshold( $threshold );
+		add_action( 'shutdown', array( $this, 'send_log_email' ) );
+	}
+
+	/**
+	 * Set handler severity threshold.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 */
+	public function set_threshold( $level ) {
+		$this->threshold = WC_Log_Levels::get_level_severity( $level );
+	}
+
+	/**
+	 * Determine whether handler should handle log.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @return bool True if the log should be handled.
+	 */
+	protected function should_handle( $level ) {
+		return $this->threshold <= WC_Log_Levels::get_level_severity( $level );
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message Log message.
+	 * @param array $context Optional. Additional information for log handlers.
+	 *
+	 * @return bool False if value was not handled and true if value was handled.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
+
+		if ( $this->should_handle( $level ) ) {
+			$this->add_log( $timestamp, $level, $message, $context );
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Send log email.
+	 *
+	 * @return bool True if email is successfully sent otherwise false.
+	 */
+	public function send_log_email() {
+		$result = false;
+
+		if ( ! empty( $this->logs ) ) {
+			$subject = $this->get_subject();
+			$body = $this->get_body();
+			$result = wp_mail( $this->recipients, $subject, $body );
+			$this->clear_logs();
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Build subject for log email.
+	 *
+	 * @return string subject
+	 */
+	protected static function get_subject() {
+		$site_name = get_bloginfo( 'name' );
+		return sprintf( __( '[%s] WooCommerce log messages', 'woocommerce' ), $site_name );
```

Comment:
How about including some more information here to allow filtering, such as the number of logs or the highest severity?